ouroboros-network/src/Ouroboros/Network/BlockFetch/ClientState.hs

```diff
@@ -249,6 +261,16 @@ newtype FetchRequest header =
         FetchRequest { fetchRequestFragments :: [ChainFragment header] }
   deriving Show
 
+-- | We can sometimes merge fetch request fragments to reduce the number of
+-- range request messages we send. We send one message per fragment. We never
+-- expect fetch requests to overlap (ie have blocks in common) but we do expect
+-- that frequently requests will \"touch\" so that two ranges could be merged
+-- into a single contigious range. This is useful because it lets us send fewer
+-- requests for bigger ranges.
+--
+-- So this semigroup instance does that style of merging. Otherwise the list
+-- of fragments is just appended.
+--
 instance HasHeader header => Semigroup (FetchRequest header) where
```

Comment:
Your new comment paragraphs are very clear. However, they also have a lot of overlap, all discussing the same concern (merging vs counting) in three different places. Maybe those places could be consolidated?

This instance, for example, could be folded into a more concrete function? Or maybe the queue (and so this instance) could explicitly track how many fragments it has merged?

These are mostly-idle suggestions, please take it or leave it :)

Reply:
Yeah, perhaps I can just refer to one of them, e.g. the semi-group instance.